api_testing/apis/orders_api.py

```diff
@@ -48,45 +49,59 @@ def create_new_order(self, **kwargs):
         :param kwargs:
         :return: response, payload
         """
+
         order_no = self.get_auto_generated_order_no()[0]['id']
-        material_type = random.choice(GeneralUtilitiesAPI().list_all_material_types()['materialTypes'])
-        article = random.choice(ArticleAPI().list_articles_by_materialtype(materialtype_id=material_type['id']))
-        testunit = random.choice(
-            TestUnitAPI().list_testunit_by_name_and_material_type(materialtype_id=material_type['id'])[0]['testUnits'])
-        testplan = random.choice(
-            ArticleAPI().list_testplans_by_article_and_materialtype(materialtype_id=material_type['id'],
-                                                                    article_id=article['id']))
+        if 'testplan' not in kwargs.keys():
+            testplan = random.choice(TestPlanAPI().get_completed_testplans())
+            material_type = testplan['materialType']
+            material_type_id = GeneralUtilitiesAPI().get_material_id(material_type)
+            article = testplan['article'][0]
+            article_api = ArticleAPI()
+            if article == "all":
+                res, _ = article_api.get_all_articles(limit=20)
+            else:
+                res, _ = article_api.quick_search_article(name=article)
+            article_id = res['articles'][0]['id']
```

Comment:
article_id is already in test plan
`article_id = testplan['articleNo'][0]`

Reply:
@ghada-37 no it's not

Reply:
ok, thanks